artifacts/otg.proto

```diff
@@ -4836,6 +4839,194 @@ message EventSubscription {
+  // 
+  // x-constraint:
+  // - /components/schemas/Port/properties/name
+  // 
+  optional string port_name = 2;
+}
+
+// Configuration for an LLDP instance
+message LldpConfig {
+
+  // Globally unique name of an object. It also serves as the primary key for arrays of
+  // objects.
+  // required = true
+  string name = 1;
+
+  // The Chassis ID is a mandatory TLV which identifies the chassis component of the endpoint
+  // identifier associated  with the transmitting LLDP agent. If mac address is specified
+  // it should be in colon seperated mac address format.
+  optional LldpConfigChassisId chassis_id = 2;
+
+  // The Port ID is a mandatory TLV which identifies the port component of the endpoint
+  // identifier associated with  the transmitting LLDP agent. If the specified port is
+  // an IEEE 802.3 Repeater port, then this TLV is optional.
+  optional LldpConfigPortId port_id = 3;
+
+  // The system name field shall contain an alpha-numeric string that indicates the system's
+  // administratively assigned  name. The system name should be the system's fully qualified
+  // domain name. If implementations support IETF RFC  3418, the sysName object should
+  // be used for this field.
+  optional LldpConfigSystemName system_name = 4;
+
+  // Specifies the amount of time in seconds a receiving device should maintain LLDP information
+  // sent  by the device before discarding it.
+  // default = 120
+  optional int32 hold_time = 5;
+
+  // Set the transmission frequency of LLDP updates in seconds.
+  // default = 30
+  optional int32 advertisement_interval = 6;
+}
+
+// The Chassis ID is a mandatory TLV which identifies the chassis component of the endpoint
+// identifier associated  with the transmitting LLDP agent. This field identifies the
+// format and source of the chassis identifier string.  It is based on the enumerator
+// defined by the LldpChassisIdSubtype object from IEEE 802.1AB MIB.
+message LldpConfigChassisId {
+
+  message Choice {
+    enum Enum {
+      unspecified = 0;
+      mac_address_subtype = 1;
+      interface_name_subtype = 2;
+      local_subtype = 3;
+    }
+  }
+  // Chassis ID subtype to be used in Chassis ID TLV.
+  // default = Choice.Enum.mac_address_subtype
+  optional Choice.Enum choice = 1;
+
+  // Description missing in models
+  optional LldpConfigChassisMacSubType mac_address_subtype = 2;
+
+  // Name of an interface of the chassis that uniquely identifies the chassis.
+  optional string interface_name_subtype = 3;
+
+  // Locally assigned name of the chassis.
+  optional string local_subtype = 4;
+}
+
+// The Port ID is a mandatory TLV which identifies the port component of the endpoint
+// identifier associated with  the transmitting LLDP agent.This field identifies the
+// format and source of the port identifier string. It is  based on the enumerator defined
+// by the PtopoPortIdType object from RFC2922.
+message LldpConfigPortId {
+
+  message Choice {
```

Comment:
is there a way to generate different message type here


normally this would just be

message LldpConfigPortId {
  enum SubType {
      SUBTYPE_UNSPECIFIED = 0;
      SUBTYPE_MAC_ADDRESS =1;
      SUBTYPE_INTERFACE_NAME = 2;
      SUBTYPE_LOCAL = 3;
}

Subtype subtype = 1;

this would be 100x more readable and 100x less confusing

Reply:
@marcushines just to understand better, is the suggestion here to define `Enum` outside `Choice` message ? The reason it's defined that way is to avoid name conflict between enum variants and message fields. 

e.g. following will result in validation error

```proto
syntax = "proto3";

message Message {
  enum Choice {
    unspecified = 0;
    variant_int = 1;
    variant_str = 2;
  }
  Choice choice = 1;
  int32 variant_int = 2;
  string variant_str = 3;
}
```

Every `Message` that includes `Choice` type, is represented as a polymorphic type in gosnappi.

This is further enforced in gosnappi like so:
```go
m := gosnappi.NewMessage()

// select variant_int as choice and set its value
m.SetVariantInt(5)
m.ToJson() // output: {"choice": "variant_int", "variant_int": 5}

// cleanup previous selection, select variant_str as choice and set its value
m.SetVariantStr("string")
m.ToJson() // output: {"choice": "variant_str", "variant_str": "string"}
```

The fact that name of the fields and that of enum variants are same, allows us to auto-generate code which asserts enum variant->field relationship.